python/pyspark/conf.py

```diff
@@ -101,13 +101,31 @@ def __init__(self, loadDefaults=True, _jvm=None, _jconf=None):
             self._jconf = _jconf
         else:
             from pyspark.context import SparkContext
-            SparkContext._ensure_initialized()
```

Comment:
Just a thought, would it be possible to just add the `--conf` args to the env variable before this is called if the jvm hasn't been created yet?  Then you wouldn't need to propagate the confs all the way through to `launch_gateway`.